src/ToEnum.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter;
+
+use BackedEnum;
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Stdlib\ArrayUtils;
+use Traversable;
+use UnitEnum;
+
+use function is_array;
+use function is_int;
+use function is_string;
+use function is_subclass_of;
+
+/**
+ * @psalm-type Options array{enum: class-string<UnitEnum>}
+ * @extends AbstractFilter<Options>
+ */
+class ToEnum extends AbstractFilter
+{
+    /** @var Options */
+    protected $options = [
+        'enum' => null,
+    ];
+
+    /**
+     * @param class-string<UnitEnum>|Traversable|Options $enumOrOptions
+     */
+    public function __construct($enumOrOptions)
```

Comment:
I would love for this to only accept `class-string<UnitEnum>` :D

Reply:
The constructor arg needs to be 

```suggestion
    public function __construct($enumOrOptions = [])
```

or

```suggestion
    public function __construct($enumOrOptions = null)
```

(null would also need to be added to the `@param` in this case)

The `FilterPluginManager` will just `new` the filter without arguments unless a factory is created _(which is pointless)_.

In order for the filter to be available in the plugin manager, `FilterPluginManager` must be updated with entries under `factories` and `aliases`

The problem is that at this point, PHPUnit will fail CI on PHP 8.0 _(Along with Psalm)_

I don't see an easy way around this at the moment unless dropping support for PHP 8.0 is an option.

Thoughts @Ocramius ?

Reply:
> The FilterPluginManager will just new the filter without arguments unless a factory is created…

Correct! 👍🏻